src/include/detail/linalg/tdb_io.h

```diff
@@ -173,24 +279,30 @@ void write_vector(
   assert(tiledb::Query::Status::COMPLETE == query.query_status());
   tiledb_helpers::submit_query(tdb_func__, uri, query);
 
-  array.close();
+  array.close();  // @note will change to unique_ptr
 }
 
 /**
  * Read the contents of a TileDB array into a std::vector.
+ *
+ * @todo Return a Vector instead of a std::vector
  */
 template <class T>
 std::vector<T> read_vector(
     const tiledb::Context& ctx,
     const std::string& uri,
     size_t start_pos,
     size_t end_pos,
-    const tiledb::TemporalPolicy temporal_policy = {}) {
```

Comment:
Why do we change the `TemporalPolicy` arguments in this PR? If we prefer `size_t` arguments for some reason, a `TemporalPolicy` should be replaced by 2 `size_t` timestamps to be able to cover both cases of `Timetravel` and `TimestampStartEnd`. This is especially important for read methods that need to implement `TimestampStartEnd` type reads.

Reply:
https://app.shortcut.com/tiledb-inc/story/39936/re-expose-temporalpolicy-and-wrap-in-python

Reply:
The current PR is equivalent to state of #173, in which Python only invokes functions from a single C++ file: index.h.
The function `ivf_index()` is the only one called from Python and the read of the `tdbColMajorMatrix` is the only place where something other than a `tiledb::TimeTravel` policy is used.  A `TemporalPolicy` interface to `tdbMatrix` is in place for this PR, and the call for creating the centroids matrix uses the `TimeStampStartEnd` policy.  So nothing differs functionally from any earlier PRs that had `TemporalPolicy` more widely used.  Properly propagating `TemporalPolicy` throughout is deferred for another PR.

In general the API needs better encapsulation of temporal [start, stop) as well as better encapsulation of spatial [start, stop).  See my comments in Story 39936, linked above.